src/steamship/invocable/mixins/indexer_pipeline_mixin.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional
+
+from steamship import File, Steamship, Task
+from steamship.invocable import PackageService, post
+from steamship.invocable.mixins.blockifier_mixin import BlockifierMixin
+from steamship.invocable.mixins.file_importer_mixin import FileImporterMixin
+from steamship.invocable.mixins.indexer_mixin import IndexerMixin
+from steamship.invocable.package_mixin import PackageMixin
+from steamship.utils.file_tags import update_file_status
+
+
+class IndexerPipelineMixin(PackageMixin):
+    """Provides a complete set of endpoints & async workflow for Document Question Answering.
+
+    This Mixin is an async orchestrator of other mixins:
+    - Importer Mixin:       to import files, e.g. YouTube videos, PDF urls
+    - Blockifier Mixin:     to convert files to Blocks -- whether that's s2t or PDF parsing, etc.
+    - Indexer Mixin:        to convert Steamship Files to embedded sharts
+
+    """
+
+    client: Steamship
+    invocable: PackageService
+    blockifier_mixin: BlockifierMixin
+    importer_mixin: FileImporterMixin
+    indexer_mixin: IndexerMixin
+
+    def __init__(self, client: Steamship, invocable: PackageService):
+        self.client = client
+        self.invocable = invocable
+
+        self.importer_mixin = FileImporterMixin(client)
+        self.invocable.add_mixin(self.importer_mixin)
+
+        self.blockifier_mixin = BlockifierMixin(client)
+        self.invocable.add_mixin(self.blockifier_mixin)
+
+        self.indexer_mixin = IndexerMixin(client)
+        self.invocable.add_mixin(self.indexer_mixin)
+
+    @post("/set_file_status")
```

Comment:
Making this a route definitely feels like an encapsulation leak

Reply:
Yeah I agree. I think there are a set of things this sprint uncovered that feel like they want to be a broader Notion design convo.

Re: the encapsulation leak side of things:

The unit of "async scheduleable task" is "public (but authed) endpoint" for us. But some async tasks are very much implementation details, like this one. I haven't yet been able to find a clean way to deal with that.